pep-9999.txt

```diff
@@ -0,0 +1,186 @@
+teaching workflow in the community, but at the same time, they create a
+barrier to entry for many. The following are a few examples of the issues
+people get into while being introduced to Python (or programming for the first
+time).
+
+- How virtual environments work is a lot of information for anyone new. It takes
+  a lot of extra time and effort to explain them.
+
+- Different platforms and shell environments require different sets of commands
+  to work on the virtual environments. Any workshop or teaching environment with
+  people coming with different operating systems installed on their laptops create a
+  lot of confusion among the participants.
+
+
+- Virtual environments need to be activated on each opened terminal, if someone
+  creates/opens a new terminal, that by default does not get the same environment
+  as in a previous terminal with virtual environment activated.
+
+
+Specification
+=============
+
+When the Python binary is executed, it attempts to determine its prefix (which
+it stores in ``sys.prefix``), which is then used to find the standard library
+and other key files, and by the ``site`` module to determine the location of the
+``site-package`` directories.  Currently the prefix is found -- assuming
+``PYTHONHOME`` is not set -- by first walking up the filesystem tree looking for
+a marker file (``os.py``) that signifies the presence of the standard library,
+and if none is found, falling back to the build-time prefix hard coded in the
+binary. The result of this process is the contents of sys.path - a list of
+locations that the Python import system will search for modules.
+
+This PEP proposes to add a new step in this process. If a ``__pypackages__``
+directory is found in the current working directory, then it will be included in
+``sys.path`` after the current working directory and just before the system
+site-packages. This way, if the Python executable starts in the given project
+directory, it will automatically find all the dependencies inside of
+``__pypackages__``.
+
+In case Python scripts, Python will try to find ``__pypackages__`` in the
+base directory of the script. If found, then it will be used, otherwise the
+Python will behave as it does currently.
+
+
+If any package management tool finds the same ``__pypackages__`` directory in
+the current working directory, it will install any packages there and also
+create it if required.
+
+
+Projects that use a source management system can include a ``__pypackages__``
+directory (empty or with e.g. a file like ``.gitignore``). After doing a fresh
+check out the source code, ``pip`` can be used to install the required
+dependencies directly into this directory.
+
+Example
+-------
+
+The following shows an example project directory structure, and different ways
+the Python executable and any script will behave.
+
+::
+
+    foo
+        __pypackages__
+            Lib
+                site-packages
+                    bottle
+        myscript.py
+
```

Comment:
Virtual environments use **lib/pythonX.Y/site-packages** inside the environment root.  While I find that overkill, wouldn't it be helpful to mirror that structure here?  Was there a reason to exclude the **pythonX.Y** path segment?

Reply:
Not on Windows they don't, and since I wrote this sample structure that's why I omitted it :)

It seems likely we'll have `__pypackages__/{abi-tag}` at the root now, which should cover that bit just fine.